src/graph/InterimResult.h

```diff
@@ -41,22 +41,37 @@ class InterimResult final {
     static Status castToBool(cpp2::ColumnValue *col);
     static Status castToStr(cpp2::ColumnValue *col);
 
+    void setColNames(std::vector<std::string> colNames) {
+        colNames_ = std::move(colNames);
+    }
+
+    void setInterim(std::unique_ptr<RowSetWriter> rsWriter);
+
+    bool hasData() const {
+        return (rsWriter_ != nullptr) || (rsReader_ != nullptr);
+    }
+
     std::shared_ptr<const meta::SchemaProviderIf> schema() const {
+        if (!hasData()) {
+            return nullptr;
+        }
         return rsReader_->schema();
```

Comment:
when rsWriter_ != nullptr, but rsReader is nullptr, the hasData return true, but rsReader is nullptr, this call will crash